### PR TITLE
Create BaseLayout component

### LINKED ISSUE #19: [Enhancement] Change pages to use Layout
### Description

Convert pages to use the Next.js [layouts](https://nextjs.org/docs/basic-features/layouts) to minimize code duplication.

This would contain the `Navbar` and `Footer` components surrounding the page contents.

### Designs

_No response_

### PR DESCRIPTION
**Pull request by gramliu**
# Description

Created a `BaseLayout` component for reusability of the common elements present on each page.

Closes #19 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/layouts/BaseLayout/index.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+interface Props {
+  children?: ReactElement | ReactElement[];
+  collapsedNavBar?: Boolean;
+}
+
+/**
+ * The base layout which includes the ScottyLabs header,
+ * a navbar, and a footer
+ */
+const BaseLayout = ({ children, collapsedNavBar = true }: Props) => {
```
Comment (hdavidethan) [inline src/layouts/BaseLayout/index.tsx]: If you're going to export this anyway, write this as a default export:
```
export default function BaseLayout(props: Props) {...}
```

File: pages/index.tsx [hunk 2]
```diff
@@ -1,28 +1,14 @@
 import type { NextPage } from "next";
-import Head from "next/head";
 import EventCarousel from "../src/components/EventCarousel";
-import Footer from "../src/components/Footer";
 import Hero from "../src/components/Hero";
-import Navbar from "../src/components/Navbar";
-import styles from "../styles/Home.module.scss";
+import BaseLayout from "../src/layouts/BaseLayout";
 
 const Home: NextPage = () => {
```
Comment (hdavidethan) [inline pages/index.tsx]: Same as the comment below

File: pages/index.tsx [hunk 3]
```diff
@@ -1,28 +1,14 @@
 import type { NextPage } from "next";
-import Head from "next/head";
 import EventCarousel from "../src/components/EventCarousel";
-import Footer from "../src/components/Footer";
 import Hero from "../src/components/Hero";
-import Navbar from "../src/components/Navbar";
-import styles from "../styles/Home.module.scss";
```
Comment (hdavidethan) [inline pages/index.tsx:7]: Does Home.module.scss still exist? And does it still need to exist?

File: src/layouts/BaseLayout/index.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import Head from "next/head";
+import { ReactElement } from "react";
+import Footer from "../../components/Footer";
+import Navbar from "../../components/Navbar";
+import styles from "./index.module.scss";
+
+interface Props {
+  children?: ReactElement | ReactElement[];
+  collapsedNavBar?: Boolean;
```
Comment (hdavidethan) [inline src/layouts/BaseLayout/index.tsx]: Boolean -> boolean

File: src/layouts/BaseLayout/index.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import Head from "next/head";
+import { ReactElement } from "react";
+import Footer from "../../components/Footer";
+import Navbar from "../../components/Navbar";
+import styles from "./index.module.scss";
+
+interface Props {
+  children?: ReactElement | ReactElement[];
```
Comment (hdavidethan) [inline src/layouts/BaseLayout/index.tsx]: I think the prescribed type for this should be `ReactNode`

See https://www.carlrippon.com/react-children-with-typescript/

File: src/layouts/BaseLayout/index.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import Head from "next/head";
+import { ReactElement } from "react";
```
Comment (hdavidethan) [inline src/layouts/BaseLayout/index.tsx]: If not doing anything special with this, can you write this as 
`import type { ReactNode } from "react";`
It's ok if not. Just trying to make sure.

**Review by hdavidethan — CHANGES_REQUESTED** [review]

File: src/layouts/BaseLayout/index.tsx (hunk 1 above)
Comment (gramliu) [reply src/layouts/BaseLayout/index.tsx]: sorry, bad habit haha I'll change this to the right format

File: src/layouts/BaseLayout/index.tsx (hunk 5 above)
Comment (gramliu) [reply src/layouts/BaseLayout/index.tsx]: TIL 💡 I've always used `ReactElement` before. Thanks

**Review by hdavidethan — APPROVED** [review]

**Pull request merged**